packages/core/content-manager/server/src/history/controllers/history-version.ts

```diff
@@ -1,13 +1,20 @@
 import { errors } from '@strapi/utils';
-import type { Common, Strapi } from '@strapi/types';
+import type { Common, Strapi, UID } from '@strapi/types';
 import { getService as getContentManagerService } from '../../utils';
 import { getService } from '../utils';
 import { HistoryVersions } from '../../../../shared/contracts';
 
 const createHistoryVersionController = ({ strapi }: { strapi: Strapi }) => {
   return {
     async findMany(ctx) {
-      if (!ctx.query.contentType || !ctx.query.documentId) {
+      const contentTypeUid = ctx.query.contentType as UID.ContentType;
+      const isSingleType = strapi.getModel(contentTypeUid).kind === 'singleType';
+
+      if (isSingleType && !contentTypeUid) {
+        throw new errors.ForbiddenError('contentType is required');
+      }
+
+      if (!contentTypeUid && !ctx.query.documentId) {
```

Comment:
isn't it necessary to have the content type uid for both single types and collection types?
Or is it that you do not store single types document ids in history?

I would say, wether we use an id in the content manager or not, single tyeps do have document ids in db, and I think it should be possible to store those here.

Could you point me where the version is stored, and what prevents storing the document id of a single type version?

Reply:
It doesn't feel like a right abstraction to require a uid or a document id, when in the end they both have ids :/ just that we don't use them in the content manager

Reply:
> isn't it necessary to have the content type uid for both single types and collection types?
I would say, wether we use an id in the content manager or not, single tyeps do have document ids in db, and I think it should be possible to store those here.

We store the contentType uid and document id for both. I believe here I am always checking for the ct uid for collection types and single types.

The version is created [here](https://github.com/strapi/strapi/blob/9f6b2ac69b7eb2e6dc1892b9e60c6684ea89b19d/packages/core/content-manager/server/src/history/services/history.ts#L57)

Examples of versions in the db look like this:

<img width="1440" alt="Screenshot 2024-02-15 at 15 55 52" src="https://github.com/strapi/strapi/assets/26598053/e6d34743-33b0-43b4-bf3d-0ecb987d7139">


My thought process was:
collectionTypes have many documents that share the same uid  'api::collection.collection' so to get history versions for a specific collectionType document I understood we would also need the `documentId`

singleTypes have a single document that share the same uid 'api::collection.collection' so I thought we don't need a documentId. 

Currently in the FE this seems to be represented the same way with the URL since when you access a collectionType you have a documentId in the url, when you access a singleType you don't.

But maybe I've misunderstood something...

Reply:
> It doesn't feel like a right abstraction to require a uid or a document id, when in the end they both have ids :/ just that we don't use them in the content manager

This is the response when fetching a collectionType in the CM. The id is the documentId, so I'm getting a bit turned around 🙃 . I believe it is the same for singleTypes. 
<img width="308" alt="Screenshot 2024-02-15 at 15 53 40" src="https://github.com/strapi/strapi/assets/26598053/4bb27901-fb5a-416e-bddb-684a554a8e78">

Reply:
>Currently in the FE this seems to be represented the same way with the URL since when you access a collectionType you have a documentId in the url, when you access a singleType you don't. But I can still get the documentId for a singleType and send it along with the request and require contentTypeUid and documentId for both cases.
But maybe I've misunderstood something...

Yep! Thats the current behaviour on the content manager, which we want to unify at some point, because it is not really necessary to have different behaviours.

Even if a single type only has a single document ID, its simpler to keep all logic the same, so we don't have to branch logic every time.

Also, Maybe I am missing some context too
```
    async findMany(ctx) {
      const contentTypeUid = ctx.query.contentType as UID.ContentType;
      const isSingleType = strapi.getModel(contentTypeUid).kind === 'singleType';

      if (isSingleType && !contentTypeUid) {
        throw new errors.ForbiddenError('contentType is required');
      }

      if (!contentTypeUid && !ctx.query.documentId) {
```

Why would you send a documentId on a find many request?

Reply:
ah 💡 because it's the find many versions of a single document, ignore my last comment

Reply:
ok, so , only to check. 
Would it be possible to send a documentId even if it's a single type? If not I believe your approach is the good one :)

Reply:
It is possible and originally I was going to do that. I opted for this approach because:

A collection-type url is like `/api::kitchensink.kitchensink/documentId` , but a singleType url is just `/api::homepage.homepage`. Currently for collectionTypes we get the documentId from the path param. When we go to the history page we push onto the route history so for collection type that looks like:

`api::kitchensink.kitchensink/documentId/history?plugins[i18n]...`

I had a couple of ideas to do this on the FE:

1. Check before navigating if it’s a singleType, if it is get the documentId from the response data and add a query param when navigating `/api::homepage.homepage/history?documentId=123`. The con is on the history page I need to check if we are working with a query param or a path param for some things. I don’t like getting the same data from two different places.
2. Always get the documentId from the same place (the response) and add it as a query param `?documentId=123`. The con is for collection-types you end up with a url like `/api::kitchensink.kitchensink/123/history?documentId=123` whereas for singleTypes you have a url like `/api::homepage.homepage/history`

In the end the backend felt like the cleanest solution, but lmk what you think

Reply:
I see :) yes, you are right. 
Seeing how things are done atm this might be the best approach!